CDCgov/prime-reportstream
# 15864 Spring Cloud Gate configuration with Swagger

This PR updates the Auth service to use Spring Cloud Gateway configuration to forward requests rather than a wildcard request mapping. It also enabled swagger when running the application locally.

Test Steps:
To ensure Gateway is working as expected
1. Run `gradle quickRun` to start up ReportStream
2. Run `gradle submissions:bootRun`
3. Run `gradle auth:bootRun`
     (Contact me if having an issue with a missing secret and I will share the staging okta secret with you)
4. Run Arnej token script to retrieve access token
    (If this is a huge pain I can send you a valid access token)
6. Make a request to POST http://localhost:9000/api/v1/reports
7. If you get a 415 Unsupported Media Type response it made it to submissions

To see Swagger
1. Run `gradle auth:bootRun`
2. Access http://localhost:9000/swagger/ui.html
3. See Swagger!

## Changes
- Removed unneeded code
- Implemented configuration based gateway pattern
- Included Swagger

## Checklist

### Testing

## Linked Issues
- Fixes #15864
